lib/tnt_api/xml_formatter.rb

```diff
@@ -16,23 +18,36 @@ def format_attrs(attrs)
 
     def format_attrs(attrs)
       attrs.each do |k, v|
-        next unless v.is_a?(String)
-        attrs[k] = escape_xml_chars(v.unicode_normalize)
+        case v.class.name
+        when "String"
+          attrs[k] = format_string(v)
+        when "Date"
+          attrs[k] = v.to_s
+        end
       end
 
-      attrs[:shipping_date] = attrs[:shipping_date].to_s
       attrs
     end
 
     private
 
-    def escape_xml_chars(str)
+    def format_string(str)
       return str unless str.is_a? String
-      str.gsub(/["&'<>]/, XML_SPECIAL_CHARACTER_MAP)
+      escape_xml_chars(
+          encode_unauthorized_chars(
+            str.unicode_normalize
+          )
+        )
+    end
+
+    def encode_unauthorized_chars(str)
```

Comment:
Do we want to do this? It's a cool way of encoding but actually it will leave us with the same issue where one character becomes 6 in a max 32 char string.

Would suggest raising an error in the case of unauthorized chars